### PR TITLE
[Based on #195 but zero-copy] Log ws message on parsing error to Message enum

### PR DESCRIPTION
**Pull request by ryo33**
As the title, this is based on #195, but zero-copy.

Also, it changes from doing `serde_json::from_slice(&data.into_data())` to doing `serde_json::from_str(&text)` of `WsMessage::Text(text)`. Other variants of `WsMessage` are handled as follows:

- `Ping` and `Pong`: Ignored
- `Bytes` and `Frames`: Ignored
- `Close`: Ends the stream

### CODE REVIEW

File: src/conn.rs [hunk 1]
```diff
@@ -132,21 +133,32 @@ impl<T: EventMessage + Unpin> Stream for Connection<T> {
...
+                        tracing::error!("Failed to deserialize WS response {}", err);
+                        Err(err.into())
+                    }
+                };
+                Poll::Ready(Some(ready))
+            }
+            Some(Ok(WsMessage::Close(_))) => Poll::Ready(None),
+            Some(Ok(WsMessage::Ping(_))) | Some(Ok(WsMessage::Pong(_))) => {
+                // ignore pings
+                Poll::Pending
```
[the remark this answers is not in the document]
Comment (ryo33) [reply src/conn.rs]: Thanks, i've fixed it!

**Comment by MOZGIII** [comment]
Is it safe to ignore the bytes and frames though?

**Review by mattsse — APPROVED** [review]

**Comment by ryo33** [comment]
> Is it safe to ignore the bytes and frames though?

Ignoring Frame is safe because we should not get it when reading a message as https://github.com/snapview/tungstenite-rs/pull/250.

IMO ignoring Bytes is also safe because cdp is a JSON-based protocol and I suppose it uses only the text type of frame.

**Comment by MOZGIII** [comment]
CDP actually can stream the responses - which would suggest using the advanced websocket capabilities - but I checked and it indeed only uses text with data embedded as base64.

See:
- https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#method-takeResponseBodyAsStream
- https://chromedevtools.github.io/devtools-protocol/tot/IO/#method-read

**Pull request merged**